pkg/cloudprovider/cloudapi/azure/azure_nsg_rules.go

```diff
@@ -520,30 +529,48 @@ func convertFromAzureSecurityRuleToInternalIngressRule(rule armnetwork.SecurityR
 	if err != nil {
 		return nil, err
 	}
+	cloudResourceID := securitygroup.CloudResourceID{
+		Name: sgName,
+		Vpc:  vnetID,
+	}
 	for _, ip := range srcIP {
-		ingressRule := securitygroup.IngressRule{
-			FromPort:  port,
-			FromSrcIP: []*net.IPNet{ip},
-			Protocol:  protoNum,
-		}
+		ingressRule := securitygroup.CloudRule{
+			Rule: &securitygroup.IngressRule{
+				FromPort:  port,
+				FromSrcIP: []*net.IPNet{ip},
+				Protocol:  protoNum,
+			},
+			AppliedToGrp: cloudResourceID.String(),
+		}
+		if desc != nil {
+			ingressRule.NetworkPolicyName = types.NamespacedName{Name: desc.Name, Namespace: desc.Namespace}.String()
+		}
+		ingressRule.Hash = ingressRule.GetHash()
 		ingressList = append(ingressList, ingressRule)
 	}
 	for _, sg := range securityGroups {
-		ingressRule := securitygroup.IngressRule{
-			FromPort:           port,
-			FromSecurityGroups: []*securitygroup.CloudResourceID{sg},
-			Protocol:           protoNum,
-		}
+		ingressRule := securitygroup.CloudRule{
+			Rule: &securitygroup.IngressRule{
+				FromPort:           port,
+				FromSecurityGroups: []*securitygroup.CloudResourceID{sg},
+				Protocol:           protoNum,
+			},
+			AppliedToGrp: cloudResourceID.String(),
+		}
+		if desc != nil {
+			ingressRule.NetworkPolicyName = types.NamespacedName{Name: desc.Name, Namespace: desc.Namespace}.String()
+		}
+		ingressRule.Hash = ingressRule.GetHash()
 		ingressList = append(ingressList, ingressRule)
 	}
 
 	return ingressList, nil
 }
 
 // convertFromAzureSecurityRuleToInternalEgressRule converts azure rules to securitygroup.EgressRule.
-func convertFromAzureSecurityRuleToInternalEgressRule(rule armnetwork.SecurityRule,
```

Comment:
Function name should also be changed now, since it is now converting to CloudRule?
may be convertFromAzureSecurityRuleToInternalEgressCloudRule